src/createTokenBridgePrepareTransactionRequest.ts

```diff
@@ -0,0 +1,92 @@
+import { Address, PublicClient, encodeFunctionData } from 'viem';
+
+import { tokenBridgeCreator } from './contracts';
+import { createTokenBridgeDefaultGasLimit, createTokenBridgeDefaultValue } from './constants';
+import { validParentChainId } from './types/ParentChain';
+import { createTokenBridgeGetInputs } from './createTokenBridge-ethers';
+import { publicClientToProvider } from './ethers-compat/publicClientToProvider';
+import { isCustomFeeTokenChain } from './utils/isCustomFeeTokenChain';
+
+type GasOverrideOptions = {
+  minimum?: bigint;
+  percentIncrease?: bigint;
+};
+
+export type TransactionRequestGasOverrides = {
+  gasLimit?: GasOverrideOptions;
+  retryableTicketFees?: GasOverrideOptions;
+};
+
+export async function createTokenBridgePrepareTransactionRequest({
+  params,
+  parentChainPublicClient,
+  childChainPublicClient,
```

Comment:
Another convention to decide. We sometimes use parentChain/orbitChain and other times we use parentChain/childChain. I personally like the former, but we definitely need to be consistent with either.

Reply:
We can rename to Orbit